content/docs/1.10.1/important-notes/_index.md

```diff
@@ -38,6 +41,103 @@ For the full release note, see [here](https://github.com/longhorn/longhorn/relea
     - [V2 Data Engine Replica Rebuild QoS](#v2-data-engine-replica-rebuild-qos)
     - [V2 Data Engine Volume Expansion](#v2-data-engine-volume-expansion)
 
+## Warning
+
+While the Longhorn v1.8.x to v1.9.x upgrade process typically migrates resources automatically to the new v1beta2 version, it is recommended to perform manual CRD storage version migration before upgrading Longhorn to v1.10.x. This is because certain operations such as etcd restore or CRD restore may cause legacy v1beta1 data to remain, and a manual migration ensures that all Longhorn custom resources are properly migrated to `v1beta2`.
+
+In addition, it is required to remove the `v1beta1` from the CRD stored version to complete the migration.
+
+For more details, see [Kubernetes official document for CRD storage version](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version).
+
+### Recommended Action Before Upgrading Longhorn To v1.10.x
+
+After Longhorn v1.9.x is installed, before upgrading to v1.10.x, perform the following manual CRD storage version migration script:
+
+```bash
+# Temporarily disable the CR validation webhook to allow updating read-only settings CRs.
+kubectl patch validatingwebhookconfiguration longhorn-webhook-validator \
+  --type=merge \
+  -p "$(kubectl get validatingwebhookconfiguration longhorn-webhook-validator -o json | \
+  jq '.webhooks[0].rules |= map(if .apiGroups == ["longhorn.io"] and .resources == ["settings"] then
+    .operations |= map(select(. != "UPDATE")) else . end)')"
+
+# Migrate CRDs that ever stored v1beta1 resources
+migration_time="$(date +%Y-%m-%dT%H:%M:%S)"
+crds=($(kubectl get crd -l app.kubernetes.io/name=longhorn -o json | jq -r '.items[] | select(.status.storedVersions | index("v1beta1")) | .metadata.name'))
+for crd in "${crds[@]}"; do
+  echo "Migrating ${crd} ..."
+  for name in $(kubectl -n longhorn-system get "$crd" -o jsonpath='{.items[*].metadata.name}'); do
+    # Attach additional annotations to trigger v1beta1 resource updating in the latest storage version.
+    kubectl patch "${crd}" "${name}" -n longhorn-system --type=merge -p='{"metadata":{"annotations":{"migration-time":"'"${migration_time}"'"}}}'
+  done
+  # Clean up the stored version in CRD status
+  kubectl patch crd "${crd}" --type=merge -p '{"status":{"storedVersions":["v1beta2"]}}' --subresource=status
```

Comment:
Are all operations, including resource migration from beta1 to beta2) and the removal of beta1 resources from stored versions, synchronous? I want to confirm that concurrency is not a concern.